### PR TITLE
bump Java.Interop/f44a11f6 and use its cecil library

### PR DESCRIPTION
**Pull request by radekdoulik**
Updated all the projects to use the `Xamarin.Android.Cecil` library
instead of `Mono.Cecil` nuget package.

### CODE REVIEW

File: Xamarin.Android.sln [hunk 1]
```diff
@@ -91,12 +91,16 @@ Project("{9344BDBB-3E7F-41FC-A0DD-8665D75EE146}") = "bundle", "build-tools\bundl
...
 EndProject
+Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "Xamarin.Android.Cecil", "external\Java.Interop\src\Xamarin.Android.Cecil\Xamarin.Android.Cecil.csproj", "{15945D4B-FF56-4BCC-B598-2718D199DD08}"
+EndProject
 Global
 	GlobalSection(SolutionConfigurationPlatforms) = preSolution
 		Debug|AnyCPU = Debug|AnyCPU
 		Release|AnyCPU = Release|AnyCPU
 		XAIntegrationDebug|AnyCPU = XAIntegrationDebug|AnyCPU
 		XAIntegrationRelease|AnyCPU = XAIntegrationRelease|AnyCPU
+		XAIntegrationDebug|Any CPU = XAIntegrationDebug|Any CPU
```
Comment (jonpryor) [inline Xamarin.Android.sln]: This and the following line shouldn't be added.

File: Xamarin.Android.sln [hunk 2]
```diff
@@ -460,6 +476,7 @@ Global
 		{0DE278D6-000F-4001-BB98-187C0AF58A61} = {E351F97D-EA4F-4E7F-AAA0-8EBB1F2A4A62}
 		{1640725C-4DB8-4D8D-BC96-74E688A06EEF} = {E351F97D-EA4F-4E7F-AAA0-8EBB1F2A4A62}
 		{7CE69551-BD73-4726-ACAA-AAF89C84BAF8} = {E351F97D-EA4F-4E7F-AAA0-8EBB1F2A4A62}
+		{15945D4B-FF56-4BCC-B598-2718D199DD08} = {04E3E11E-B47D-4599-8AFC-50515A95E715}
```
Comment (jonpryor) [inline Xamarin.Android.sln]: Perhaps this should go into the `Tools` folder? Not sure it's immediately relevant for `Core`.

`Tools` is `864062D3-A415-4A6F-9324-5820237BA058`.

File: src/Xamarin.Android.Tools.Aidl/Xamarin.Android.Tools.Aidl.csproj [hunk 3]
```diff
@@ -58,4 +46,10 @@
   <ItemGroup>
     <None Include="packages.config" />
   </ItemGroup>
+  <ItemGroup>
+    <ProjectReference Include="..\..\external\Java.Interop\src\Xamarin.Android.Cecil\Xamarin.Android.Cecil.csproj">
```
Comment (jonpryor) [inline src/Xamarin.Android.Tools.Aidl/Xamarin.Android.Tools.Aidl.csproj]: Ditto.

**Comment by jonpryor** [comment]
The PR [failed to build](https://jenkins.mono-project.com/job/xamarin-android-pr-builder/282/consoleText):

    Linker/MonoDroid.Tuner/Linker.cs(8,18): error CS0234: The type or namespace name `Mdb' does not exist in the namespace `Mono.Cecil'. Are you missing an assembly reference?

Plus 15 other errors.

**Comment by radekdoulik** [comment]
Thanks for the review. I have fixed the issues and made it build.

**Pull request merged**